### ISSUE TITLE
Spell check is crashing the program

### ISSUE THREAD

**Issue by v-i-o-l-e-t** [olivierkes/manuskript#273]
Hey there!

I just downloaded Manuskript about 20 minutes ago and am loving all of it, but I've had some problems with spell check.

-It's enabled, but not checking words. I've tried just typing random letters and the program doesn't underline any of them or bring up an option to fix when I right-click them.
-I tried pressing F9 to manually check and the program crashed
-I also tried disabling spell check, but the program crashed. When I started again it was still enabled.
-The italics, bold and underline buttons also don't work, but if I type the formatting for them directly in the box, everything is fine. This probably isn't related, but thought I would bring it up.

I'm running version 0.6.0 on Windows 10 (if you need more information, I'll try to find it. Not much of a computer person.)

It's not a huge problem as I'll export to Word when I'm finished but I thought you should know.

Thanks in advance!

Edit:
Okay, so, I went into full screen and enabled spell check, and now it's working in full screen, no crashing at all. I should have mentioned earlier that when trying to enable spell check, I was in the fiction layout, no full screen. I'm assuming spell check only working in full screen is not a feature, but if it is, sorry for creating a bug thread.

**Comment by v-i-o-l-e-t** [comment]
Okay, so, I went into full screen and enabled spell check, and now it's working in full screen, no crashing at all. I should have mentioned earlier that when trying to enable spell check, I was in the fiction layout, no full screen. I'm assuming spell check only working in full screen is not a feature, but if it is, sorry for creating a bug thread.

**Comment by gedakc** [comment]
Thank you @v-i-o-l-e-t for your interesting in Manuskript.

Would you be able to provide a screen shot or a sample of the text that demonstrates the spell check problem?

The spell checker *should* work in both normal and full screen modes.

Following is a screen shot illustrating spell check working on Manuskript 0.6.0 in Windows 10.

![manuskript-0 6 0-win10-spellcheck](https://user-images.githubusercontent.com/10405019/34170001-81454e62-e4a6-11e7-9a0d-9c9949e60ea2.png)

**Comment by v-i-o-l-e-t** [comment]
Well, this is embarrassing, but it's working now. I went to take a screenshot and tried enabling it once more and there's no problem. Both the normal and full screen view are highlighting the misspelled word. I haven't changed anything since I reported this, so I'm not sure what's changed. I can still provide screenshots, if you'd like.

Also, I should point out that italics, bold and underline buttons are still not working, but at this rate it looks like if I just wait a bit it'll fix itself, haha.

**Comment by gedakc** [comment]
No worries @v-i-o-l-e-t.  Strange things can and do occur.  If it happens again then please report the issue along with relevant details -- ideally the steps that will permit us to recreate the problem.

Regarding the format buttons for bold, italics, and underline, these have been broken since Manuskript was changed to use Markdown instead of Rich Text format.  In the latest git **develop** branch these buttons have been removed.  Hence the buttons will be gone from the next release (0.7.0).  You can read more about this in [Issue 59 - Format buttons in window not working](https://github.com/olivierkes/manuskript/issues/59).

I am closing this issue since we are unable to recreate the spell check not working problem.

@v-i-o-l-e-t thanks again for your interest in helping us to improve Manuskript.  Please feel free to re-open this report if the spell check problem recurs.

**Comment by gedakc** [comment]
@olivierkes, is it possible that the spell check on/off setting still uses the older method?

By older method I recall you had to change the handling of some other boolean settings so that these worked on all OSes. 

Anyways it's just a thought.

**Comment by gedakc** [comment]
Following is a description of my investigation of the spelling check crash problem.

To simulate starting Manuskript for the first time on Windows 10 I used **regedit** to remove the **HKEY_Current_User/Software/manuskript** entry.

I started manuskript 0.6.0, created a novel, and typed some text into the editor.  When I looked at the menu **Tools -> Dictionary** I noticed that no dictionary was selected.

![manuskript-win10-no-dict-selected](https://user-images.githubusercontent.com/10405019/34266604-0dec9424-e637-11e7-9ecb-6788f6f8a977.png)

When I selected **Tools -> Spellcheck** manuskript crashed.

If I start manuskript 0.6.0 again, create another document, and select **Tools -> Spellcheck** again, manuskript crashes again.  Note that I have not yet selected selected a dictionary.

If I start manuskript 0.6.0, **open an existing document**, and select **Tools -> Spellcheck** then manuskript **does not** crash, even though I have not yet selected a dictionary.

The spelling check crash also occurs in the current git **develop** branch.

I tried to recreate this problem on kubuntu 16.04, but it does not occur for me because the **en_CA** English dictionary is always selected, even on initial run (I removed **~/.local/share/manuskript**, **~/.config/manuskriptrc**, and **~/.config/manuskript**).

**Possible Solution**:  To avoid the crash, we might consider only enabling spellcheck if a dictionary is selected, or always selecting at least one of the available dictionaries.

Hopefully this additional information will help to identify the root cause of this issue. 

@olivierkes if you need any more information then please feel free to ask and I can run more tests.

**Comment by olivierkes** [comment]
Thanks @gedakc for the thorough report.

I think this is not connected to the way boolean are stored, but to how manuskript determines the default dict.

What's the error message?

**Comment by gedakc** [comment]
> What's the error message?

'Sorry about that @olivierkes.  Even though I'm a programmer, I sometimes forget some of the essential information.  :-)

Following is the console output when manusrkript crashes after selecting **Tools -> Spellcheck** when there is no dictionary set.
```
C:\manuskript-0.6.0-win32>manuskript.exe
Debug: Web rendering engine used: QWebView
Running manuskript version 0.6.0.
Note: No translator found or loaded for locale en_US.
Project C:/Users/gedak/Documents/deleteme-1.msk saved.
Traceback (most recent call last):
  File "manuskript\mainWindow.py", line 1259, in toggleSpellcheck
  File "manuskript\ui\views\textEditView.py", line 412, in toggleSpellcheck
  File "site-packages\enchant\__init__.py", line 558, in __init__
  File "site-packages\enchant\__init__.py", line 168, in __init__
  File "site-packages\enchant\__init__.py", line 565, in _init_this
  File "site-packages\enchant\__init__.py", line 310, in _request_dict_data
  File "site-packages\enchant\__init__.py", line 259, in _raise_error
enchant.errors.DictNotFoundError: invalid tag character found

C:\manuskript-0.6.0-win32>
```

**Comment by gedakc** [comment]
Today I tested this issue on Windows XP and was able to re-create the crash.  I have identified the cause and will create a PR to fix the issue.

The error log on Windows XP was as follows.  Note that I do not have an **English_Canada**, **en_CA**, or **Canadian English** dictionary installed for Enchant.

```
C:\manuskript.git>c:\Python34\python.exe bin\manuskript
Debug: Web rendering engine used: QWebView
Running manuskript version 0.6.0.
Note: No translator found or loaded for locale en_CA.
Project deleteme-3.msk saved.
Traceback (most recent call last):
  File "C:\manuskript.git\bin\..\manuskript\mainWindow.py", line 1313, in toggleSpellcheck
    w.toggleSpellcheck(val)
  File "C:\manuskript.git\bin\..\manuskript\ui\views\textEditView.py", line 438, in toggleSpellcheck
    self._dict = enchant.Dict(self.currentDict)
  File "c:\Python34\lib\site-packages\enchant\__init__.py", line 558, in __init__
    _EnchantObject.__init__(self)
  File "c:\Python34\lib\site-packages\enchant\__init__.py", line 168, in __init__
    self._init_this()
  File "c:\Python34\lib\site-packages\enchant\__init__.py", line 565, in _init_this
    this = self._broker._request_dict_data(self.tag)
  File "c:\Python34\lib\site-packages\enchant\__init__.py", line 310, in _request_dict_data
    self._raise_error(eStr % (tag,),DictNotFoundError)
  File "c:\Python34\lib\site-packages\enchant\__init__.py", line 258, in _raise_error
    raise eclass(default)
enchant.errors.DictNotFoundError: Dictionary for language 'English_Canada' could not be found

C:\manuskript.git>
```

**Comment by gedakc** [comment]
The spell check crash fix contained in pull request #303 was merged into the **develop** branch for inclusion in the next release of Manuskript.
